inc/saimacsec.h

```diff
@@ -345,117 +345,136 @@ typedef enum _sai_macsec_port_attr_t
 } sai_macsec_port_attr_t;
 
 /**
- * @brief MACsec port counter IDs in sai_get_macsec_stats() call
+ * @brief Attribute Id for sai_macsec_flow
  */
-typedef enum _sai_macsec_port_stat_t
+typedef enum _sai_macsec_flow_attr_t
 {
     /**
-     * @brief Malformed packets dropped before MACsec processing, not in 802.1ae MIB
+     * @brief Start of MACsec Flow attributes
      */
-    SAI_MACSEC_PORT_STAT_PRE_MACSEC_DROP_PKTS,
+    SAI_MACSEC_FLOW_ATTR_START,
 
     /**
-     * @brief Packets classified as control packets for MACsec processing, not in 802.1ae MIB
+     * @brief MACsec direction
+     *
+     * @type sai_macsec_direction_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
      */
-    SAI_MACSEC_PORT_STAT_CONTROL_PKTS,
+    SAI_MACSEC_FLOW_ATTR_MACSEC_DIRECTION = SAI_MACSEC_FLOW_ATTR_START,
 
     /**
-     * @brief Packets classified as data packets for MACsec processing, not in 802.1ae MIB
+     * @brief MACsec port object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_MACSEC_PORT
      */
-    SAI_MACSEC_PORT_STAT_DATA_PKTS,
+    SAI_MACSEC_FLOW_ATTR_PORT_ID,
 
     /**
-     * @brief IEEE 802.1ae defined ifOutOctets or ifInOctets for MACSEC uncontrolled port
+     * @brief List of MACsec ACL entries associated with this flow.
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_ACL_ENTRY
      */
-    SAI_MACSEC_PORT_STAT_OCTETS_UNCONTROLLED,
+    SAI_MACSEC_FLOW_ATTR_ACL_ENTRY_LIST,
```

Comment:
Why not use the match list directly without hinting ACLs. ACLs might have scale impact for large number of flow and there might be an implementation that uses an exact match table or any other way.

Reply:
ACL entry is rule match table associated with flow. This list indicates number of ACL rule entry attached to this MACsec flow. ACL entry are rule match table it can be attached to any flow.